src/particle.c

```diff
@@ -17,6 +17,10 @@ particle particle_init(int pid, int charge, double beta, int status, int sector,
     // Inherent vars.
     p.is_valid = true;
     p.is_trigger_electron = (pid == 11 && status < 0);
+    p.is_hadron = false;
+    if(!p.is_trigger_electron){
```

Comment:
This if statement is serves no purpose --- the for loop below filters out electrons anyway!

Reply:
Yeah, I think this "if" would make sense if the process inside its scope was computationally expensive.